### PR TITLE
Add Delete task command

### LINKED ISSUE #23: As a user, I can delete a task
... so that I can remove unwanted tasks or tasks that were entered incorrectly

### PR DESCRIPTION
**Pull request by amresh2000**
- Add command to delete a task: `DeleteTaskCommand`
- Add tests for new class

Closes #23

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/task/AddTaskCommandTest.java [hunk 1]
```diff
@@ -199,7 +204,7 @@ public boolean hasTask(Task task) {
     /**
      * A Model stub that always accept the person being added.
      */
-    private class ModelStubAcceptingTaskAdded extends ModelStub {
+    protected class ModelStubAcceptingTaskAdded extends ModelStub {
```
Comment (kianseong) [inline src/test/java/seedu/address/logic/commands/task/AddTaskCommandTest.java]: Why are we changing this to protected?

**Review by kianseong — APPROVED** [review]
Organised and follows OOP principles!

LGTM!

File: src/main/java/seedu/address/logic/commands/task/DeleteTaskCommand.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public static final String MESSAGE_USAGE = COMMAND_WORD_FULL
+            + ": Deletes the task that index number refers to.\n"
+            + "Parameters: \n"
+            + "Example: " + COMMAND_WORD + " 1";
+    public static final String MESSAGE_SUCCESS = "Task deleted: %1$s";
+
+    private final Index targetIndex;
+
+    /**
+     * Creates an DeleteCommand to remove the specified {@code Task}
```
Comment (yilinzyl) [inline src/main/java/seedu/address/logic/commands/task/DeleteTaskCommand.java]: DeleteTaskCommand

File: src/main/java/seedu/address/logic/parser/task/DeleteTaskCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new DeleteTaskCommand object
+ */
+public class DeleteTaskCommandParser implements Parser<DeleteTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteCommand
+     * and returns an DeleteCommand object for execution.
```
Comment (yilinzyl) [inline src/main/java/seedu/address/logic/parser/task/DeleteTaskCommandParser.java:21]: DeleteTaskCommand

File: src/main/java/seedu/address/logic/parser/task/DeleteTaskCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteTaskCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args);
+
+        Index index = TaskParserUtil.parseIndex(argMultimap.getPreamble());
```
Comment (yilinzyl) [inline src/main/java/seedu/address/logic/parser/task/DeleteTaskCommandParser.java]: In other commands, we caught the ParseException from parseIndex and throw a new ParseException containing String.format(MESSAGE_INVALID_COMMAND_FORMAT, XXXCommand.MESSAGE_USAGE) that will be displayed to the user. You can consider following the same format for consistency of our UI!

**Review by yilinzyl — COMMENTED** [review]
Some tiny things to note, but overall LGTM!

**Review by yilinzyl — APPROVED** [review]

**Pull request merged**